token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyService.java

```diff
@@ -36,8 +36,10 @@ public String retrieveTokenKeys(URI tokenKeysEndpointUri) throws OAuth2ServiceEx
 			if (statusCode == HttpStatus.SC_OK) {
 				return bodyAsString;
 			} else {
-				throw OAuth2ServiceException
-						.createWithStatusCodeAndResponseBody("Error retrieving token keys", statusCode, bodyAsString);
+				throw OAuth2ServiceException.builder("Error retrieving token keys")
+						.withStatusCode(statusCode)
```

Comment:
i would have expected 
.withUri(tokenKeysEndpointUri)

Reply:
Thanks I missed that one!